AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1515,40 +1515,36 @@ - (void)_removeSubnode:(ASDisplayNode *)subnode
   [subnode __setSupernode:nil];
 }
 
+// NOTE: You must not called this method while holding the receiver's property lock. This may cause deadlocks.
 - (void)removeFromSupernode
 {
   ASDisplayNodeAssertThreadAffinity(self);
-  BOOL shouldRemoveFromSuperviewOrSuperlayer = NO;
+  _propertyLock.lock();
+    __weak ASDisplayNode *supernode = _supernode;
+    __weak UIView *view = _view;
+    __weak CALayer *layer = _layer;
+    BOOL layerBacked = _flags.layerBacked;
+  _propertyLock.unlock();
```

Comment:
nit: can we use a scoped locker still? So:
```
{
    ASDN::MutexLocker l(_propertyLock);
    __weak ASDisplayNode *supernode = _supernode;
    __weak UIView *view = _view;
    __weak CALayer *layer = _layer;
    BOOL layerBacked = _flags.layerBacked;
}
```

The lockers are exception-safe through RAII semantics, while manual lock/unlock will remain deadlocked if an exception is thrown between the calls (for instance, if something insane happens inside layerBacked). Probably not an issue here, but worth preserving the safety for the future.